src/main/java/org/jabref/gui/maintable/SmartConstrainedResizePolicy.java

```diff
@@ -22,75 +23,158 @@
+
+        // Calculate newWidth based on delta and constraint of the column
+        double oldWidth = column.getWidth();
+        double newWidth;
+        if (delta < 0) {
+            double minWidth = column.getMinWidth();
+            LOGGER.trace("MinWidth {}", minWidth);
+            newWidth = Math.max(minWidth, oldWidth + delta);
+        } else {
+            double maxWidth = column.getMaxWidth();
+            LOGGER.trace("MaxWidth {}", maxWidth);
+            newWidth = Math.min(maxWidth, oldWidth + delta);
+        }
+        LOGGER.trace("Size: {} -> {}", oldWidth, newWidth);
+        if (oldWidth == newWidth) {
+            return Optional.empty();
         }
+        return Optional.of(newWidth);
     }
 
+    /**
+     * Resizes the table based on the content. The main driver is that if the content might fit into the table without horizontal scroll bar.
+     * In case the content fitted before the resize and will fit afterwards, the delta is distributed among the remaining columns - instead of just moving the columns right of the current column.
+     * In case the content does not fit anymore, a horizontal scroll bar is shown.
+     * In all cases the minimum/maximum width of each column is respected.
+     */
     private Boolean constrainedResize(TableView.ResizeFeatures<?> prop) {
         TableView<?> table = prop.getTable();
+        Double tableWidth = getContentWidth(table);
         List<? extends TableColumnBase<?, ?>> visibleLeafColumns = table.getVisibleLeafColumns();
-        return constrainedResize(prop,
-                false,
-                getContentWidth(table) - 2,
-                visibleLeafColumns);
+        Double requiredWidth = visibleLeafColumns.stream().mapToDouble(TableColumnBase::getWidth).sum();
+        Double delta = prop.getDelta();
+
+        TableColumn<?, ?> userChosenColumnToResize = prop.getColumn();
+
+        boolean columnsCanFitTable = requiredWidth + delta < tableWidth;
+        if (columnsCanFitTable) {
+            LOGGER.debug("User shrunk column in that way thus that window can contain all the columns");
+            if (requiredWidth >= tableWidth) {
+                LOGGER.debug("Before, the content did not fit. Rearrange everything.");
+                return rearrangeColumns(table);
+            }
+            LOGGER.debug("Everything already fit. We distribute the delta now.");
+
+            // Content does already fit
+            // We "just" need to readjust the column widths
+            determineNewWidth(userChosenColumnToResize, delta)
+                    .ifPresent(newWidth -> {
+                        double currentWidth = userChosenColumnToResize.getWidth();
+                        Double deltaToApply = newWidth - currentWidth;
+                        for (TableColumnBase<?, ?> col : visibleLeafColumns) {
+                            Double share = expansionShare.get(col);
```

Comment:
Shouldn't a resize of a column not also update the expansion share directory? The user prefers to give more space to a certain column...

Reply:
We restore the shares now after a column was resized completely (when another column is resized)